### PR TITLE
Feature/batch adjustment

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR adds some additional configuration and rewiring to the single-cell analysis control workflow to deal with batch effect adjustment:

 - Derive batch fields from IDF file at the config step and store in the config file. I also switched the cell type field to use the same mechanism (having previously had it derived via a separate Nextflow process.
 - Prevent re-quantification from being triggered automatically by changes in the configuration. This is no longer desirable when there are tertiary-specific changes like the batch field in the configuration. 
 - The batch-enabled Galaxy workflow now also produces 'project' h5ad files with merged experimental outputs, so I've added some handling for that file to get bundled.

This paired with the associated changes in the tertiary workflow https://github.com/ebi-gene-expression-group/scxa-workflows/pull/20.

### CODE REVIEW

File: bin/utils.sh
Comment (pcm32) [inline bin/utils.sh]: Unfortunately the MAGE-TAB spec is quite lenient on this and you could expect to find different capitalisations, maybe this replacement could include an ignore case term? So that you accept things like "characteristics". Also, to add to the fun, I understand that the space in "Factor Value" is optional, hehehe..

File: main.nf
Comment (pcm32) [inline main.nf:328]: is it fine that CONF_BY_EXP_SPECIES_FOR_TERTIARY is repeated in this enum?

File: main.nf
Comment (pcm32) [inline main.nf:673]: A glorious and expected addition!

File: main.nf
Comment (pcm32) [inline main.nf:1159]: I'm assuming that this is the only place were we are calling filterSinglets.sh (given its changes of arguments earlier on this PR).

**Review by pcm32 — APPROVED** [review]
LGTM in general, just a few remarks, feel free to merge when those are sorted/confirmed.

File: bin/utils.sh
Comment (pinin4fjords) [reply bin/utils.sh]: That one is optional too????

File: main.nf
Comment (pinin4fjords) [reply main.nf:328]: Huh, weird that this didn't cause issues. But will tidy- thanks.

File: main.nf
Comment (pinin4fjords) [reply main.nf:1159]: Yes. It's only necessary to keep Scanpy happy here.

File: bin/utils.sh
Comment (pinin4fjords) [reply bin/utils.sh]: Corrected regex accordingly

**Comment by pinin4fjords** [comment]
I need to merge this along with https://github.com/ebi-gene-expression-group/scxa-workflows/pull/20, so will wait until that's reviewed and merged.

**Comment by pinin4fjords** [comment]
Thanks for review @pcm32

**Pull request merged**
